modules/web3/hooks/useGasPrice.ts

```diff
@@ -0,0 +1,18 @@
+import useSWR from 'swr';
+import { fetchGasPrice } from '../helpers/fetchGasPrice';
+
+type GasResponse = {
+  data?: number | string | undefined;
+  loading: boolean;
+  error?: Error;
+};
+
+export const useGasPrice = (): GasResponse => {
+  const { data, error } = useSWR('fetch-gas', () => fetchGasPrice('fast'));
```

Comment:
Our default `refreshInterval` for SWR is set to 60 seconds, so if we want it any quicker than that, we should specify it here an as option. I think we could maybe increase to 10 or 15 seconds, but probably not a huge deal. Wdyt?

Reply:
I initially was thinking it should be on the slower side since most of our transactions don't really need to be gas competitive, but I'd be open to speeding it up. Another open question I meant to pose to everybody would be which speed setting we should default to. I chose 'fast' because it was kind of a middle ground.

https://www.etherchain.org/api/gasnow